libkbfs/conflict_resolver.go

```diff
@@ -195,8 +206,18 @@ func (cr *ConflictResolver) Resolve(unmerged MetadataRevision,
 		return
 	}
 
+	ci := conflictInput{unmerged, merged}
+	func() {
+		cr.inputLock.Lock()
+		defer cr.inputLock.Unlock()
+		// Cancel any running CR before we return, so the caller can be
+		// confident any ongoing CR superseded by this new input will be
+		// canceled before it releases any locks it holds.
+		_ = cr.cancelExistingLocked(ci)
```

Comment:
Would it make sense to return immediately here if `ci` isn't valid, instead of letting the background goroutine do it?